### PR TITLE
[CI][GHActions] Introduce GitHub Actions Iroha1

### PR DESCRIPTION
**Pull request by kuvaldini**
### Description of the Change

Partially clozes #948

### Benefits

Checks with GitHub Actions

### CODE REVIEW

File: .github/build-iroha1.src.yml
Comment (BulatSaif) [inline .github/build-iroha1.src.yml:75]: Add Comment or TODO why this is disabled

File: .github/build-iroha1.src.yml
Comment (BulatSaif) [inline .github/build-iroha1.src.yml:284]: same

File: .github/github-runner-on-aws-ec2-spot/.gitignore
Comment (BulatSaif) [inline .github/github-runner-on-aws-ec2-spot/.gitignore:3]: You are working in a team, use `Backend Type: s3`,  don't store tfstate locally.

**Review by BulatSaif — COMMENTED** [review]
fix check: `Jenkins: x86_64 Linux Debug gcc9 `

File: .github/build-iroha1.src.yml
Comment (kuvaldini) [inline .github/build-iroha1.src.yml:27]:
```suggestion
    branches: [ main, support/1.* ]
```

File: .github/build-iroha1.src.yml
Comment (kuvaldini) [inline .github/build-iroha1.src.yml:30]:
```suggestion
    branches: [ main, support/1* ]  ## target branches
```

File: .github/build-iroha1.src.yml
Comment (i1i1) [inline .github/build-iroha1.src.yml:136]: Humb1t is not maintainer anymore, so we should change secrets.

Comment (kuvaldini) [reply .github/build-iroha1.src.yml:136]: Very good point. Thank you.

File: .github/build-iroha1.src.yml
Comment (kuvaldini) [reply .github/build-iroha1.src.yml:82]: done

File: .github/build-iroha1.src.yml
Comment (kuvaldini) [reply .github/build-iroha1.src.yml:295]: done

File: .github/github-runner-on-aws-ec2-spot/.gitignore
Comment (kuvaldini) [reply .github/github-runner-on-aws-ec2-spot/.gitignore:3]: to be disscussed with voice

File: .github/build-iroha1.src.yml
Comment (kuvaldini) [reply .github/build-iroha1.src.yml:143]: removed

File: .github/build-iroha1.src.yml
Comment (BulatSaif) [inline .github/build-iroha1.src.yml:50]: add a small job to check what content of  `.github/build-iroha1.src.yml` and `.github/workflows/build-iroha1.yml`  match.
People can forget to add a pre-commit hook

File: docker/iroha-builder/Dockerfile
Comment (i1i1) [inline docker/iroha-builder/Dockerfile:49]: Do you squash docker images afterwards? Because there is no need otherwise to cleanup apt cache as it will be still in docker image layers.

If no, you can merge this layers together.

Comment (BulatSaif) [reply docker/iroha-builder/Dockerfile:49]: You can delete this step, this looks like a mistake.

File: .github/build-iroha1.src.yml
Comment (kuvaldini) [reply .github/build-iroha1.src.yml:50]: Nice Idea. Thank you. TBD

File: docker/iroha-builder/Dockerfile
Comment (kuvaldini) [reply docker/iroha-builder/Dockerfile:49]: Best practices say this should be https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#run

Sure, you have a ratio to merge all apt-get steps in one. And definitely sure clean step should be a part of install step. This is to be fixed.

I intentionally separated packages into different steps to be able to reuse layers when one of them is changed. I.e. do not reinstall all the packages listed. But take previous layer with most packages installed and add yet another ones. This is a kind of optimization when change Dockerfile often. 

After all job is done, I believe I should merge all apt steps in one. But if someone will add just one package, the whole layer will be rebuilt. I will take much more time and traffic. And this is a point to discuss.
But from the other side, this is a rare procedure - rebuild docker image.

Comment (kuvaldini) [reply docker/iroha-builder/Dockerfile:49]: I do not think we should squash layers. At least base Ubuntu should be a separate layer.

Comment (kuvaldini) [reply docker/iroha-builder/Dockerfile]: All apt-get merged in one step.

File: .github/build-iroha1.src.yml
Comment (kuvaldini) [reply .github/build-iroha1.src.yml:50]: Done. Works fine

**Comment by kuvaldini** [comment]
@BulatSaif Could you please take a look ones more and approve this PR

**Comment by BulatSaif** [comment]
> @BulatSaif Could you please take a look ones more and approve this PR

I will check it tomorrow morning, meanwhile, can you fix failed builds, I see it has problems cleaning VM.

**Comment by BulatSaif** [comment]
I reviewed PR,  big work done!

Did we achieved goals?
1. It is faster than Jenkins ci.
2. It is easy to maintain by developers.
---
1. I couldn't check the first one because the build is failing currently.

2. As I see the level of complexity remains the same (my subjective opinion), I believe that more developers know YAML than groovy, but in your code, you used advanced YAML (Anchors and Aliases) which negates all the simplicity of YAML. You can reduce the complexity by splitting action into multiple files and adding more documentation on how to edit and test GH Actions. But if it satisfies the developers I am fine with what. 

---
I will approve pr when my previous comments are  applied: 

> 1. can you fix failed builds, I see it has problems cleaning VM.
> 2. please comment line 213 and 223 in https://github.com/hyperledger/iroha/blob/main/Jenkinsfile to disable yellow status

**Comment by kuvaldini** [comment]
@BulatSaif 

1. Yes it is faster ~= 22min
изображение

1.a. Build currently is falling due to problem in runner configuration, I notified @ravenolf 

2. At the moment the only developer works on CI side is @kuvaldini, others are aware how to work the that workflow but not deeply involved

File: Jenkinsfile
Comment (BulatSaif) [inline Jenkinsfile:227]: The wrong line, you need to disable in  `'Commit in Open PR':` section  (4 line up)

Comment (kuvaldini) [reply Jenkinsfile]: done

**Review by BulatSaif — APPROVED** [review]
GBTPR (God bless this PR)

**Review by kamilsa — APPROVED** [review]

**Comment by kuvaldini** [comment]
no reaction from CI

**Pull request merged**
